### PR TITLE
deps: update protobuf dependencies to v3.23.1 and gprc to 1.55.1

### PR DESCRIPTION
**Pull request by mpeddada1**
TODO: Close https://github.com/googleapis/sdk-platform-java/pull/1369 after this PR is merged.

### CODE REVIEW

**Review by suztomo — APPROVED** [review]

**Comment by suztomo** [comment]
You may have to update the C++ option. https://github.com/googleapis/sdk-platform-java/pull/1369/files#r1164950006

**Comment by suztomo** [comment]
ci / showcase is failing.

```
  bazel-bin/showcase/verify_proto
(14:45:41) INFO: Elapsed time: 341.420s, Critical Path: 31.43s
(14:45:41) INFO: 625 processes: 219 internal, 388 linux-sandbox, 18 worker.
(14:45:41) INFO: Build completed successfully, 625 total actions
(14:45:41) INFO: Running command line: bazel-bin/showcase/verify_proto proto
diff -ru /home/runner/work/sdk-platform-java/sdk-platform-java/showcase/proto-gapic-showcase-v1beta1/src/main/java/com/google/showcase/v1beta1/AttemptSequenceRequest.java /home/runner/.cache/bazel/_bazel_runner/cf5fb8c2c66b8a944ea6af7c9b248db5/execroot/gapic_generator_java/bazel-out/k8-fastbuild/bin/showcase/verify_proto.runfiles/gapic_generator_java/proto_unpacked/proto-google-cloud-showcase-v1beta1-java/src/main/java/com/google/showcase/v1beta1/AttemptSequenceRequest.java
--- /home/runner/work/sdk-platform-java/sdk-platform-java/showcase/proto-gapic-showcase-v1beta1/src/main/java/com/google/showcase/v1beta1/AttemptSequenceRequest.java	2023-05-23 14:38:00.460649223 +0000
+++ /home/runner/.cache/bazel/_bazel_runner/cf5fb8c2c66b8a944ea6af7c9b248db5/execroot/gapic_generator_java/bazel-out/k8-fastbuild/bin/showcase/verify_proto.runfiles/gapic_generator_java/proto_unpacked/proto-google-cloud-showcase-v1beta1-java/src/main/java/com/google/showcase/v1beta1/AttemptSequenceRequest.java	1980-01-01 00:00:00.000000000 +0000
@@ -26,11 +26,6 @@
     return new AttemptSequenceRequest();
   }
 
-  @java.lang.Override
-  public final com.google.protobuf.UnknownFieldSet
-  getUnknownFields() {
-    return this.unknownFields;
-  }
   public static final com.google.protobuf.Descriptors.Descriptor
       getDescriptor() {
     return com.google.showcase.v1beta1.SequenceOuterClass.internal_static_google_showcase_v1beta1_AttemptSequenceRequest_descriptor;
@@ -191,11 +186,13 @@
     return com.google.protobuf.GeneratedMessageV3
         .parseWithIOException(PARSER, input, extensionRegistry);
   }
+
   public static com.google.showcase.v1beta1.AttemptSequenceRequest parseDelimitedFrom(java.io.InputStream input)
       throws java.io.IOException {
     return com.google.protobuf.GeneratedMessageV3
         .parseDelimitedWithIOException(PARSER, input);
   }
```

=> https://github.com/googleapis/sdk-platform-java/tree/main/showcase#update-the-golden-showcase-files

@mpeddada1 Can you update the golden file?

File: gax-java/dependencies.properties [hunk 1]
```diff
@@ -23,7 +23,7 @@ version.gax_httpjson=0.112.1-SNAPSHOT
 # Versions for dependencies which actual artifacts differ between Bazel and Gradle.
 # Gradle build depends on prebuilt maven artifacts, while Bazel build depends on Bazel workspaces
 # with the sources.
-version.com_google_protobuf=3.21.12
+version.com_google_protobuf=3.23.1
 version.google_java_format=1.15.0
 version.io_grpc=1.54.0
```
Comment (burkedavison) [inline gax-java/dependencies.properties:28]: This version should be updated.

Comment (mpeddada1) [reply gax-java/dependencies.properties]: Thank you! Done.

**Review by suztomo — APPROVED** [review]

**Comment by suztomo** [comment]
Sonar is complaining https://github.com/googleapis/sdk-platform-java/pull/1698/checks?check_run_id=13697463444

> 0.0% Coverage on New Code (is less than 80%)

But we're not adding new code. It's fine.

**Pull request merged**